controller/appcontroller.go

```diff
@@ -1606,6 +1639,7 @@ func (ctrl *ApplicationController) autoSync(app *appv1.Application, syncStatus *
 		return nil
 	}
 
+	hasMultipleSources := app.Spec.Sources != nil && len(app.Spec.Sources) > 0
```

Comment:
Noted that you added a HasMultipleSources() method, maybe use that here instead?